perfkitbenchmarker/linux_virtual_machine.py

```diff
@@ -708,3 +731,127 @@ def SetupProxy(self):
 
     if commands:
       self.RemoteCommand(";".join(commands))
+
+
+class ContainerizedDebianMixin(DebianMixin):
+  """Class representing a Virtual Machine that runs Remote
+  Commands within a Docker Container running Ubuntu"""
+
+  def PrepareVMEnvironment(self):
+    """Initializes docker before proceeding with preparation."""
+    self.RemoteHostCommand('mkdir -p %s' % vm_util.VM_TMP_DIR)
+    self.Install('docker')
+    self.InitDocker()
+    super(ContainerizedDebianMixin, self).PrepareVMEnvironment()
+
+  def InitDocker(self):
+    """Initializes the docker container daemon."""
+    init_docker_cmd = ('sudo docker run -d '
+                       '--net=host '
+                       '--workdir=%s '
+                       '-v %s:%s ' % (CONTAINER_WORK_DIR,
+                                      vm_util.VM_TMP_DIR,
+                                      CONTAINER_MOUNT_DIR))
+    for sd in self.scratch_disks:
+      init_docker_cmd += '-v %s:%s ' % (sd.mount_point, sd.mount_point)
```

Comment:
Try to avoid loops where you accumulate strings using += - instead create a list and join them. See https://google-styleguide.googlecode.com/svn/trunk/pyguide.html?showone=Strings#Strings